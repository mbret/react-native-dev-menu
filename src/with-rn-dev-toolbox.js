/**
 * @flow
 */
import React from 'react'
import { RNDevToolboxContext } from './RNDevToolbox'

export const withRNDevToolbox = (Component: any) => {
  class ComponentWithRNDevToolbox extends React.Component<*> {
    render () {
      return (
        <RNDevToolboxContext.Consumer>
          {rnDevToolbox => {
            return (
              <Component
                {...this.props}
                reactNativeDevToolbox={rnDevToolbox}
              />
            );
          }}
        </RNDevToolboxContext.Consumer>
      )
    }
  }

  return ComponentWithRNDevToolbox
}