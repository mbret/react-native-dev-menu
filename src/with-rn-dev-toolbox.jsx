/* global $Diff */
/**
 * @flow
 */
import React, { Component } from 'react'
import { RNDevToolboxContext } from './RNDevToolboxContext'
import hoistStatics from 'hoist-non-react-statics'
import type { ComponentType } from 'react'
import type { RNDevToolboxInterface } from './RNDevToolboxBase'

/**
 * Inject the dev tool instance into the component.
 * @see https://flow.org/en/docs/react/hoc/
 */
export function withRNDevToolbox<Props: {}> (
  BaseComponent: ComponentType<Props>
): ComponentType<$Diff<Props, { rnDevToolbox: RNDevToolboxInterface | void }>> {
  class ComponentWithRNDevToolbox extends Component<*> {
    render () {
      return (
        <RNDevToolboxContext.Consumer>
          {rnDevToolbox => {
            return (
              <BaseComponent
                {...this.props}
                rnDevToolbox={rnDevToolbox}
              />
            )
          }}
        </RNDevToolboxContext.Consumer>
      )
    }
  }

  return hoistStatics(ComponentWithRNDevToolbox, BaseComponent)
}
