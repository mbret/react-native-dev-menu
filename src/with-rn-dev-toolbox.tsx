/* eslint no-unused-vars:0 */
/**
 *
 */
import React, { Component, ComponentType } from 'react'
import { RNDevToolboxContext } from './RNDevToolboxContext'
import hoistStatics from 'hoist-non-react-statics'
import { RNDevToolboxInterface } from './RNDevToolboxBase'

type OwnProps = { rnDevToolbox?: RNDevToolboxInterface | void }

/**
 * Inject the dev tool instance into the component.
 * @see https://flow.org/en/docs/react/hoc/
 */
export function withRNDevToolbox<P> (
  BaseComponent: ComponentType<P & OwnProps>
): ComponentType<P> {
  class ComponentWithRNDevToolbox extends Component<P & OwnProps> {
    render () {
      return (
        <RNDevToolboxContext.Consumer>
          {rnDevToolbox => {
            return (
              <BaseComponent
                {...this.props}
                rnDevToolbox={rnDevToolbox}
              />
            )
          }}
        </RNDevToolboxContext.Consumer>
      )
    }
  }

  return hoistStatics(ComponentWithRNDevToolbox, BaseComponent)
}
