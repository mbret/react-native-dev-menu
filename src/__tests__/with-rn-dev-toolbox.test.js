/* eslint-env jest */
/**
 * @flow
 */
import React from 'react'
import renderer from 'react-test-renderer'
import { withRNDevToolbox } from '../with-rn-dev-toolbox'
import { Text } from 'react-native'

describe('withRNDevToolbox', () => {
  it('should return component', () => {
    const enhanced = withRNDevToolbox(Text)

    expect(typeof enhanced).toBe('function')
  })

  test('renders correctly', () => {
    const Enhanced = withRNDevToolbox(Text)
    const a = renderer.create(<Enhanced />).toJSON()
    expect(a).toMatchSnapshot()
  })
})
