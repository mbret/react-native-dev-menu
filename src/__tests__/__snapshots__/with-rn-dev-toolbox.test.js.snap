// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRNDevToolbox renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  rnDevToolbox={Object {}}
/>
`;
